packages/search-token-lambda/functions/token/ensureTokenGenerated.ts

```diff
@@ -0,0 +1,71 @@
+require('isomorphic-fetch');
+require('abortcontroller-polyfill');
+
+import {PlatformClient, RestUserIdType} from '@coveord/platform-client';
+
+export async function ensureTokenGenerated() {
+  const platform: PlatformClient = new PlatformClient({
+    /**
+     * The Plaform URL to use.
+     * https://platform.cloud.coveo.com is the default platform host.
+     * However, you can target a different environment by changing the host value.
+     *
+     * Example:
+     * Use "https://platformhipaa.cloud.coveo.com" if you want to target the HIPAA environment.
+     *
+     * You can also target a different region (e.g. https://platform-au.cloud.coveo.com)
+     * See https://docs.coveo.com/en/2976/coveo-solutions/deployment-regions-and-strategies#data-residency
+     */
+    host: process.env.PLATFORM_URL,
+    /**
+     * The unique identifier of your Coveo organization.
+     * To retrieve your org ID, see https://docs.coveo.com/en/148/manage-an-organization/retrieve-the-organization-id
+     */
+    organizationId: process.env.ORGANIZATION_ID,
+    /**
+     * An API key with the impersonate privilege in the target organization.
+     * See https://docs.coveo.com/en/1718/manage-an-organization/manage-api-keys#add-an-api-key
+     */
+    accessToken: process.env.API_KEY!,
+  });
+
+  try {
+    const {token} = await platform.search.createToken({
+      /****** Mandatory parameters ******/
+      /**
+       * The security identities to impersonate when authenticating a query with this search token.
+       * The userIds array should contain at least one security indentity.
+       * See https://docs.coveo.com/en/56/#userids-array-of-restuserid-required
+       */
+      userIds: [
+        {
+          name: process.env.USER_EMAIL!,
```

Comment:
The way the cli works with this package & the search token server one is the following:
- you login with the cli to your coveo account on an org
- you create a ui (e.g. atomic) using the CLI, this will bootstrap a project and set a few env variables according to you login info
- the new project copies either the token node server or the lambda function into the projet as a **starting point** with comments and doc.
- Once the user as this set up, he can go ahead & improve upon the basic use case

Reply:
@louis-bompart  @y-lakhdar do I understand this correctly?

Reply:
> Once the user as this set up, he can go ahead & improve upon the basic use case

I don't see how one can adapt the code for production as along as we're using environment variables?

One can copy-paste code, but this means the npm package is not usable on it's own? One needs to open `node-modules`?

And if so, what's the benefit of publishing an independent package? Why not bootstrap a project with a cli and copy-paste the generated file instead of needing to dig into `node_modules`?

Atm, is this the dev experience for prod?
1. `npm i search-token-lambda`
2. Copy files from `node_modules`
3. `npm rm search-token-lambda`
4. `npm i @coveord/platform-client`


cc: @olamothe for this [comment](https://github.com/coveo/cli/pull/558#issuecomment-976841103).

Reply:
The users are not copy pasting the code: When they will run the `ui:atomic` command they will get a local project that contains this code you are seeing here, already "setup properly" for them (correct env variable).

They don't need to copy paste anything from npm module.

They can then decide to modify the code of their project however they see fit.

Reply:
Using the cli makes sense, but then why publish an independent npm package with the code?
Are we imagining devs running `npm i search-token-lambda`? If so, how would one use and extend it?

The current design seems limited to one email address, which is fine for a demo, but too limited for a standalone npm package?

Reply:
Putting it in a separate package helps using it standalone. It is possible to simply do `npm init @coveo/atomic` without even going through the CLI, which facilitates the process.

Reply:
To answer the mention, yes. The CLI is not in the business of being an archwizard, just a wizard if I may jest.
By that, I mean that we do not expect the user to push their stuff in production straight out from the CLI.

The goal of the lambda is to provide an easier deployment for demo and dev purposes exclusively.
The reasons are that we want users to be sure of what they're doing, and the best way for the CLI to do it is to remove the stupid stuff, but always leave the user in control and be extremely transparent about what's going on.

For why another package. Ideally, we want 'atomic' packages (like each package does 'one thing' only), with the notable exception of the CLI. (and even that is subject to evolution with pluginification of the CLI).
This allows power users to cherry-pick what they want with `npm init @coveo/atomic`.

Reply:
Alright, thanks for clarifying. I think it could be helpful to provide this context to developers in `README.md`, noting that
- The package is designed for demo deployments by default, and
- Explaining how to use it as a starting point via `npm init`.

Thanks!

Reply:
Good point raised @samisayegh , we should indeed be clear about what we do here in our readmes